hs_app_timeseries/models.py

```diff
@@ -281,20 +559,100 @@ def metadata(self):
+        # to the resource
+        if self.has_sqlite_file:
+            return False
+        if not self.has_csv_file:
+            return False
+
+        return True
+
+    @property
+    def can_update_sqlite_file(self):
+        # guard property to determine when the sqlite file can be updated as result of
+        # metadata changes
+        return self.has_sqlite_file and self.metadata.has_all_required_elements()
 
     @classmethod
     def get_supported_upload_file_types(cls):
-        # final phase of this resource type implementation will support 3 file types
-        # return (".csv", ".xml", ".sqlite")
-        # phase-1 of implementation supports only sqlite file
-        return (".sqlite",)
+        # either a csv or a sqlite file can be uploaded
+        # the internal storage format will always be sqlite
+        # if a csv file is uploaded, a sqlite file will be generated using data
+        # from the uploaded csv file.
+        # the original uploaded csv file will also be kept as part of the resource
+        return ".sqlite", ".csv"
 
     @classmethod
-    def can_have_multiple_files(cls):
-        # can have only 1 file
+    def allow_multiple_file_upload(cls):
+        # can upload only 1 file
+        # however, the resource can have a sqlite file and a csv file
+        # if the user uploads a sqlite file, then the resource will have
+        # only one file (the uploaded sqlite file).
+        # if the user uploads a csv file, a sqlite file will be created and
+        # both the uploaded csv file and the generated sqlite file will be part
+        # of the resource.
         return False
 
+    @classmethod
+    def can_have_multiple_files(cls):
+        # can have a csv file and a sqlite file - one of those is uploaded the other one is system
+        # generated
+        return True
+
+    def has_required_content_files(self):
+        # sqlite file is the required content file
+        return self.has_sqlite_file
+
+    def add_blank_sqlite_file(self, user):
+        # add the blank SQLite file to this resource (self)
+        if not self.can_add_blank_sqlite_file:
+            err_msg = """Can't add blank ODM2 SQLite file to the resource.
+             Resource may already have a SQLite file or there may not be enough metadata
+             for the resource."""
+            raise Exception(err_msg)
+
+        log = logging.getLogger()
+
+        # copy the blank sqlite file to a temp directory
+        temp_dir = tempfile.mkdtemp()
+        odm2_sqlite_file_name = 'ODM2.sqlite'
+        odm2_sqlite_file = 'hs_app_timeseries/files/{}'.format(odm2_sqlite_file_name)
+        target_temp_sqlite_file = os.path.join(temp_dir, odm2_sqlite_file_name)
+        shutil.copy(odm2_sqlite_file, target_temp_sqlite_file)
+
+        # add the sqlite file to the resource
+        files = [UploadedFile(file=open(target_temp_sqlite_file, 'rb'),
+                              name=odm2_sqlite_file_name)]
```

Comment:
Is the copy to temp directory really needed here? If there is no change to the blank ODM2.sqlite, wouldn't it be sufficient to directory upload this file rather than doing a copy beforehand?